packages/scoutgame/src/builderNfts/refreshScoutNftBalance.ts

```diff
@@ -0,0 +1,64 @@
+import type { BuilderNftType } from '@charmverse/core/prisma-client';
+import { prisma } from '@charmverse/core/prisma-client';
+import { findOrCreateWalletUser } from '@packages/users/findOrCreateWalletUser';
+import type { Address } from 'viem';
+
+import { getPreSeasonTwoBuilderNftContractReadonlyClient } from './clients/preseason02/getPreSeasonTwoBuilderNftContractReadonlyClient';
+import { getBuilderNftStarterPackReadonlyClient } from './clients/starterPack/getBuilderContractStarterPackReadonlyClient';
+import { builderNftChain } from './constants';
+
+export async function refreshScoutNftBalance({
+  wallet,
+  tokenId,
+  contractAddress,
+  nftType
+}: {
+  wallet: Address;
+  tokenId: number;
+  contractAddress: Address;
+  nftType: BuilderNftType;
+}) {
+  const existingBuilderNft = await prisma.builderNft.findFirstOrThrow({
+    where: {
+      tokenId,
+      contractAddress: {
+        equals: contractAddress,
+        mode: 'insensitive'
+      }
+    }
+  });
+
+  const balance = await (nftType === 'starter_pack'
+    ? getBuilderNftStarterPackReadonlyClient({
```

Comment:
suggestion maybe for later: we could probably just one one simple client interface since we just need to call getBalance. you could probably use a standard client for ERC1155 even.